libafl_nyx/src/helper.rs

```diff
@@ -0,0 +1,97 @@
+/// [`NyxHelper`] is used to wrap `NyxProcess`
+use std::{
+    fmt::{self, Debug},
+    path::Path,
+};
+
+use libnyx::{NyxProcess, NyxReturnValue};
+
+pub type NyxResult<T> = Result<T, String>;
+
+pub struct NyxHelper {
+    pub nyx_process: NyxProcess,
+    /// real size of trace_bits
+    pub real_map_size: usize,
+    // real size of the trace_bits
+    pub map_size: usize,
+    /// shared memory with instruction bitmaps
+    pub trace_bits: *mut u8,
+}
+
+const MAX_FILE: u32 = 1024 * 1024;
+#[derive(Clone, Copy)]
+pub enum NyxProcessType {
+    /// stand alone mode
+    ALONE,
+    /// parallel mode's parent, used to create snapshot
+    PARENT,
+    /// parallel mode's child, consume snapshot and execute
+    CHILD,
+}
+impl NyxHelper {
+    /// create `NyxProcess` and do basic settings
+    pub fn new(
+        target_dir: &Path,
+        cpu_id: u32,
+        snap_mode: bool,
+        nyx_type: NyxProcessType,
+    ) -> NyxResult<Self> {
+        let sharedir = target_dir.to_str().unwrap();
```

Comment:
please avoid unwrap() if you can.
you can change this new()'s return type to Result<X, Error> then change all unwrap-s to ? operator

Reply:
I use `expect` instead of `?`, because `to_str()` returns `Option` and not supporting `?`